lib/workers/repository/config-migration/branch/index.ts

```diff
@@ -1,31 +1,49 @@
+import is from '@sindresorhus/is';
+import type { ConfigMigrationResult } from '..';
 import { GlobalConfig } from '../../../../config/global';
 import type { RenovateConfig } from '../../../../config/types';
 import { logger } from '../../../../logger';
 import type { FindPRConfig, Pr } from '../../../../modules/platform';
 import { platform } from '../../../../modules/platform';
-import { ensureComment } from '../../../../modules/platform/comment';
 import { scm } from '../../../../modules/platform/scm';
 import { getMigrationBranchName } from '../common';
 import { ConfigMigrationCommitMessageFactory } from './commit-message';
 import { createConfigMigrationBranch } from './create';
 import type { MigratedData } from './migrated-data';
 import { rebaseMigrationBranch } from './rebase';
 
+interface CheckConfigMigrationBranchResult extends ConfigMigrationResult {
+  migrationBranch?: string;
+}
+
 export async function checkConfigMigrationBranch(
   config: RenovateConfig,
   migratedConfigData: MigratedData | null,
-): Promise<string | null> {
+): Promise<CheckConfigMigrationBranchResult> {
   logger.debug('checkConfigMigrationBranch()');
   if (!migratedConfigData) {
     logger.debug('checkConfigMigrationBranch() Config does not need migration');
-    return null;
+    return {};
+  }
+
+  if (
+    !config.configMigration &&
+    (is.undefined(config.dependencyDashboardChecks?.configMigrationInfo) ||
+      config.dependencyDashboardChecks?.configMigrationInfo === 'no-checkbox' ||
+      config.dependencyDashboardChecks?.configMigrationInfo === 'unchecked')
```

Comment:
this logic is hard to read/maintain

Reply:
Maybe nest inside `if (!config.configMigration)` and then have a separate `if` for the other three inside